tools/gosec/gosec.go

```diff
@@ -0,0 +1,111 @@
+package main
+
+import (
+	"encoding/xml"
+	"flag"
+	"io/ioutil"
+	"log"
+	"sort"
+	"strconv"
+	"strings"
+)
+
+// These structs are copied from gosec
+type junitXMLReport struct {
+	XMLName    xml.Name    `xml:"testsuites"`
+	Testsuites []testsuite `xml:"testsuite"`
+}
+
+type testsuite struct {
+	XMLName   xml.Name   `xml:"testsuite"`
+	Name      string     `xml:"name,attr"`
+	Tests     int        `xml:"tests,attr"`
+	Testcases []testcase `xml:"testcase"`
+}
+
+type testcase struct {
+	XMLName xml.Name `xml:"testcase"`
+	Name    string   `xml:"name,attr"`
+	Failure failure  `xml:"failure"`
+}
+
+type failure struct {
+	XMLName xml.Name `xml:"failure"`
+	Message string   `xml:"message,attr"`
+	Text    string   `xml:",innerxml"`
+}
+
+var (
+	junitFile  string
+	outputFile string
+)
+
+func init() {
+	flag.StringVar(&junitFile, "junit", "junit-gosec.xml", "usage")
+	flag.StringVar(&outputFile, "output", "", "usage")
+}
+
+func parseJunit(filename string) *junitXMLReport {
+	log.Printf("Reading data from %s", filename)
+	data, err := ioutil.ReadFile(filename)
+	if err != nil {
+		log.Fatalf("Failed to read report, %v", err)
+	}
+
+	var junit junitXMLReport
+	xml.Unmarshal(data, &junit)
+	return &junit
+}
+
+func main() {
```

Comment:
IIUC we are sorting the JUnit results here by name (and then by line number), right? If the gosec tool itself doesn't do this,  we might be better off with opening a PR on it (and use our changed fork in the meantime).

I'd expect the tool to use sorting by severity rather than by name?

Reply:
Gosec does sort the issues by severity. But that's not enough since the individual test cases can change order. Let me attemt to do a PR  for gosec & switch to fork meanwhile.